### ISSUE TITLE
`continue` with label is not seen

### ISSUE THREAD

**Issue by samueltardieu** [rust-lang-nursery/rust-clippy#1990]
The following code (produced after encountering the problem in real code) shows that clippy fails to see that `continue 'label` might cause the labelled loop to loop (if `a` is `true`).

```rust
pub fn f(a: bool, b: bool) {
    'label: loop {
        loop {
            if a {
                continue 'label;
            }
            if b {
                break;
            }
        }
        break;
    }
}
```

```
warning: this loop never actually loops
  --> src/lib.rs:2:5
   |
2  | /     'label: loop {
3  | |         loop {
4  | |             if a {
5  | |                 continue 'label;
...  |
11 | |         break;
12 | |     }
   | |_____^
   |
   = note: #[warn(never_loop)] on by default
   = help: for further information visit https://github.com/rust-lang-nursery/rust-clippy/wiki#never_loop
```

**Comment by mcarton** [comment]
Should be quite easy, we are probably just matching a `None` somewhere we should just match a `_` for the label of a loop.
